packages/pyroscope-flamegraph/src/FlameGraph/FlameGraphRenderer.tsx

```diff
@@ -232,8 +232,9 @@ class FlameGraphRenderer extends React.Component<
   };
 
   isSameFlamebearer = (prevFlame: Flamebearer, currFlame: Flamebearer) => {
-    // This is a poor heuristic, but it should work most of the times
-    return prevFlame.numTicks === currFlame.numTicks;
+    // TODO: come up with a less resource intensive operation
```

Comment:
I think a better solution would be to generate a random ID for each response on the server side.